src/pcre2test.c

```diff
@@ -5071,15 +5071,14 @@ switch(cmd)
     {
     while (isspace(*argptr)) argptr++;
     if (*argptr == 0) break;
-    for (i = 1; i < sizeof(newlines)/sizeof(char *); i++)
+    for (uint16_t j = 1; j < sizeof(newlines)/sizeof(char *); j++)
```

Comment:
Do we use such declaration? Btw is this a bug?

Reply:
C99 style declarations in loops are in the codebase for almost two years (see #163), not sure what you refer as a bug though.

Reply:
The variable 'i' is declared as it seems. If its value needs to be preserved, then this code is wrong.

Reply:
The variable `i` is declared as a `size_t` at the beginning of `process_command()` and used in multiple places (mainly as a loop variable) but in none of those places it was meant to be preserved AFAIK.

Reply:
It looks like an unsigned variable is perfect here. `sizeof(newlines)/sizeof(char *)` is already an unsigned value. The patch also removes a `PCRE2_ASSERT`

Reply:
> The patch also removes a `PCRE2_ASSERT`

yes, as it is mentioned in the commit message, since it is not necessary and was introduced by mistake (instead of an static assert that would break at build time, and that was suggested instead) in #466.

note that any truncation should be also visible in the clang DEV job so implementing an static assert macro (which I plan to do anyway) and use it here is not that urgent.